### PR TITLE
add new tutorial for synthetic biology: basic assembly analysis

### PR DESCRIPTION
**Pull request by kenza12**
Hi, this is a first PR for creating a tutorial in a new topic 'synthetic-biology' for basic assembly analysis. We would like to run this tutorial in https://galaxy-synbiocad.org/ .
Thank you

### CODE REVIEW

File: CONTRIBUTORS.yaml [hunk 1]
```diff
@@ -14,14 +14,21 @@
...
 
 # two dummy entries so that autogenerated tutorials do not fail to build
-contributor1:
-    name: Contributor A (Please add contributors to the CONTRIBUTORS.yaml file)
-    halloffame: "no"
-    joined: 2020-06
-contributor2:
-    name: Contributor B (Please list all contributors of this tutorial in the metadata section)
-    halloffame: "no"
-    joined: 2020-06
```
Comment (hexylena) [inline CONTRIBUTORS.yaml:24]: Please re-add these, these should not be removed

File: CONTRIBUTORS.yaml [hunk 2]
```diff
@@ -14,14 +14,21 @@
...
+    name: Thomas Duigou
+    email: thomas.duigou@inrae.fr
+
+breakthewall:
+    name: Joan Hérisson
+    email: joan.herisson@univ-evry.fr
+
+guillaume-gricourt:
+    name: Guillaume Gricourt
+    email: guillaume.gricourt@inrae.fr
```
Comment (hexylena) [inline CONTRIBUTORS.yaml:31]: Could you please insert the entries where they belong alphabetically in the full list?

File: topics/synthetic-biology/slides/introduction.html [hunk 3]
```diff
@@ -0,0 +1,14 @@
+---
+layout: introduction_slides
```
Comment (hexylena) [inline topics/synthetic-biology/slides/introduction.html:2]: Should this file be deleted?

File: topics/synthetic-biology/metadata.yaml [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+name: synthetic-biology
+type: use
+title: Title of the topic
+summary: Summary of the topic
+requirements:
+- type: internal
+  topic_name: introduction
+docker_image: ''
+maintainers:
+- maintainers
```
Comment (hexylena) [inline topics/synthetic-biology/metadata.yaml:11]: This file needs to be filled out, the maintainer list should include some names (perhaps all of yours), and the title/summary should be filled out with some text.

File: topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.md [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+time_estimation: 15 min
+key_points:
+- Annotation of SBML pathway using Selenzyme
+- Generating Constructs using rpbasicdesign
+- Generating python scripts using dnabot
+contributors:
+- Kenza BAZI KABBAJ
+- Thomas DUIGOU
+- Joan HERISSON
+- Guillaume GRICOURT
```
Comment (hexylena) [inline topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.md:22]: These need to be replaced with the keys you used in the contributors.yaml file, e.g. `kenza12` instead of the name.

File: topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.md [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+title: Designing plasmids encoding predicted pathways by using the BASIC assembly method
+zenodo_link: https://zenodo.org/record/6123887#.YhTCPhvjJH4
+questions:
+- How to design plasmids encoding predicted pathways using BASIC assembly method ?
+- How to generate scripts enabling the automatic build of plasmids ?
+objectives:
+- Annotate the SBML pathway using Selenzy tool.
+- Design plasmids encoding predicted pathways using rpbasicdesign tool.
+- Generate dnabot scripts enabling the automatic build of plasmids.
+time_estimation: 15 min
```
Comment (hexylena) [inline topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.md:13]:
```suggestion
time_estimation: 15M
```

File: topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.md [hunk 7]
```diff
@@ -0,0 +1,201 @@
+---
+layout: tutorial_hands_on
+
+title: Designing plasmids encoding predicted pathways by using the BASIC assembly method
+zenodo_link: https://zenodo.org/record/6123887#.YhTCPhvjJH4
```
Comment (hexylena) [inline topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.md:5]:
```suggestion
zenodo_link: https://zenodo.org/record/6123887
```

File: topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.md [hunk 8]
```diff
@@ -0,0 +1,201 @@
...
+
+---
+
+
+# Introduction
+{:.no_toc}
+
+Synthetic biology is a novel engineering discipline which requires computational tools for the design of metabolic pathways for the production of chemicals.
+
+In this tutorial, we will use a set of tools from the **Genetic Design - BASIC Assembly Workflow** (https://galaxy-synbiocad.org) which will enable you to design plasmids implementing metabolic pathways for the bioproduction of lycopene in _E.coli_ using the [BASIC assembly method](https://doi.org/10.1021/sb500356d), and to generate scripts enabling the automatic build of these plasmids using an Opentrons liquid handler robot. After downloading these scripts onto a computer connected to an Opentrons, one can perform the automated construction of the plasmids at the bench.
```
Comment (hexylena) [inline topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.md:32]: The DOI you used here indicates this might be a good place to use a reference.
```suggestion
In this tutorial, we will use a set of tools from the **Genetic Design - BASIC Assembly Workflow** (https://galaxy-synbiocad.org) which will enable you to design plasmids implementing metabolic pathways for the bioproduction of lycopene in _E.coli_ using the BASIC assembly method (as described in {% cite ... %}), and to generate scripts enabling the automatic build of these plasmids using an Opentrons liquid handler robot. After downloading these scripts onto a computer connected to an Opentrons, one can perform the automated construction of the plasmids at the bench.
```

and then using https://doi2bib.org to get the bibtex entry for that DOI and adding it to the tutorial.bib

File: topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.md [hunk 9]
```diff
@@ -0,0 +1,201 @@
...
+>
+> 3. Rename the datasets if you want to shorten the file names.
+>
+>    {% snippet faqs/galaxy/datasets_rename.md %}
+>
+{: .hands_on}
+
+# Find enzymes using **Selenzyme**
+
+At first, a pathway generated by the Pathway Analysis workflows (https://galaxy-synbiocad.org) is provided as an SBML file to the [Selenzyme](https://doi.org/10.1093/bioinformatics/bty065) tool. Selenzyme searches for enzymes corresponding to each reaction of the pathway. It performs a reaction similarity search in the reference reaction database [MetaNetX](https://www.metanetx.org) and outputs an updated SBML file annotated with the enzyme UniProt IDs.
```
Comment (hexylena) [inline topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.md:81]: you might use a `{% cite .. %}` here as well

**Review by hexylena — COMMENTED** [review]
Some formatting/infrastructure type contents. 

Will there be more tutorials coming in the future for synthetic biology?

File: CONTRIBUTORS.yaml (hunk 1 above)
Comment (kenza12) [reply CONTRIBUTORS.yaml]: Yes, we are planning to add more tutorials for other workflows (Retrosynthesis analysis, Pathway analysis, Genetic Design (Gibson, Golden Gate, LCR)). Thank you.

File: CONTRIBUTORS.yaml (hunk 2 above)
Comment (kenza12) [reply CONTRIBUTORS.yaml]: Ok, done.

File: topics/synthetic-biology/slides/introduction.html (hunk 3 above)
Comment (kenza12) [reply topics/synthetic-biology/slides/introduction.html:2]: Yes, I'll provide slides later. Thank you.

File: topics/synthetic-biology/metadata.yaml (hunk 4 above)
Comment (kenza12) [reply topics/synthetic-biology/metadata.yaml]: Ok, Done.

File: topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.md (hunk 5 above)
Comment (kenza12) [reply topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.md]: Ok, done.

File: topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.md (hunk 6 above)
Comment (kenza12) [reply topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.md]: Ok, done.

File: topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.md (hunk 7 above)
Comment (kenza12) [reply topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.md]: Ok, Done.

File: topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.md (hunk 8 above)
Comment (kenza12) [reply topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.md]: Ok, done.

File: topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.md (hunk 9 above)
Comment (kenza12) [reply topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.md]: Ok, done.

File: topics/synthetic-biology/tutorials/basic_assembly_analysis/workflows/Genetic_Design_BASIC_Assembly.ga [hunk 10]
```diff
@@ -0,0 +1,413 @@
+{
+    "a_galaxy_workflow": "true",
```
Comment (shiltemann) [inline topics/synthetic-biology/tutorials/basic_assembly_analysis/workflows/Genetic_Design_BASIC_Assembly.ga:2]:
```suggestion
    "a_galaxy_workflow": "true",
    "tags": ["synthetic-biology"],
```

Comment (shiltemann) [reply topics/synthetic-biology/tutorials/basic_assembly_analysis/workflows/Genetic_Design_BASIC_Assembly.ga:2]: oh wait, I saw you already did that, wonder why the test is still failing, will look into it, disregard this suggestion

File: topics/synthetic-biology/tutorials/basic_assembly_analysis/workflows/Genetic_Design_BASIC_Assembly.ga [hunk 11]
```diff
@@ -0,0 +1,413 @@
...
+            "workflow_outputs": [
+                {
+                    "label": null,
+                    "output_name": "dnabot_scripts",
+                    "uuid": "ce5be4f0-48fb-45f0-84ad-15348645c950"
+                }
+            ]
+        }
+    },
+    "tags": [
```
Comment (shiltemann) [inline topics/synthetic-biology/tutorials/basic_assembly_analysis/workflows/Genetic_Design_BASIC_Assembly.ga:408]: aha, remove this `tags`, it is overriding the one above

File: CONTRIBUTORS.yaml [hunk 12]
```diff
@@ -214,6 +214,10 @@ bfranicevic:
     name: Branka Franicevic
     joined: 2021-10
 
+breakthewall:
+    name: Joan Hérisson
+    email: joan.herisson@univ-evry.fr
```
Comment (shiltemann) [inline CONTRIBUTORS.yaml:219]:
```suggestion
    email: joan.herisson@univ-evry.fr
    joined: 2022-03
```

Comment (shiltemann) [reply CONTRIBUTORS.yaml:219]: could you add a "joined" line like this to all the new contributors you added? :)

File: topics/synthetic-biology/slides/introduction.html (hunk 3 above)
Comment (shiltemann) [reply topics/synthetic-biology/slides/introduction.html]: slides would be great! This sounds like a very interesting topic, but currently not a lot of explanation for people who are new to the topic, some intro slides covering the basic concepts involved in synthetic biology would be great to have, and maybe also slides with the tutorials (or a bit more background/explanation in the tutorial itself?)
 
 great work here!

File: CONTRIBUTORS.yaml (hunk 1 above)
Comment (kenza12) [reply CONTRIBUTORS.yaml]: Hi @shiltemann , yes I'm in the GTN Gitter. Thank you for your help.

File: topics/synthetic-biology/slides/introduction.html (hunk 3 above)
Comment (kenza12) [reply topics/synthetic-biology/slides/introduction.html:2]: Yes I agree. I just added new slides for Synthetic Biology Introduction and more background/explanation in the tutorial. Thanks again.

File: topics/synthetic-biology/tutorials/basic_assembly_analysis/workflows/Genetic_Design_BASIC_Assembly.ga (hunk 11 above)
Comment (kenza12) [reply topics/synthetic-biology/tutorials/basic_assembly_analysis/workflows/Genetic_Design_BASIC_Assembly.ga]: ok, done.

File: CONTRIBUTORS.yaml (hunk 12 above)
Comment (kenza12) [reply CONTRIBUTORS.yaml:219]: ok done. I also added 2 more contributors.

File: topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.md [hunk 13]
```diff
@@ -0,0 +1,225 @@
...
+
+The workflow scheme we will use is shown below. First, we will run the steps of this workflow individually so as not to neglect the understanding of the intermediate steps as well. Then, we will run the workflow automatically so that it itself retrieves the outputs from the previous step and gives them as input to the next tool.
+
+![Genetic Design - BASIC Assembly Workflow](../../images/basic_assembly_workflow.png)
+
+Note that a video tutorial is also available for this workflow : [Galaxy BASIC Assembly Workflow Tutorial ](https://youtu.be/it_JTVdvaRo)
```
Comment (shiltemann) [inline topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.md:47]: A video, nice! Would you be ok with adding that to the [GTN Video Library](https://gallantries.github.io/video-library/)? That way the video will be automatically linked from the top of the tutorials, and from the topic page (under "recordings"). And it would also make it easier to create online training events around your tutorials.

We would re-upload this tutorial to the GalaxyProject YouTube account, and ask you to correct the auto-generated captions that YouTube generates, so that we can provide quality subtitles and increase accessibility for our users.  We will copy the autogenerated subs to a Google Doc for you and others to fix, usually its just a matter of punctuation and fixing technical terms)

**Review by shiltemann — COMMENTED** [review]
Hi @kenza12!

Thanks for your contribution here! 

There is currently a merge conflict, can you fix that? (please let me know if you need help doing this)

And I've left some minor comments in the code. Like I said I can't say much about the scientific content, I will see if anybody wants to review this, but if you've had your team test and double-check this that is ok too

File: topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.md [hunk 14]
```diff
@@ -0,0 +1,225 @@
...
+- How to generate scripts enabling the automatic build of plasmids ?
+objectives:
+- Annotate the SBML pathway using Selenzy tool.
+- Design plasmids encoding predicted pathways using rpbasicdesign tool.
+- Generate dnabot scripts enabling the automatic build of plasmids.
+time_estimation: 15M
+key_points:
+- Annotation of SBML pathway using Selenzyme
+- Generating Constructs using rpbasicdesign
+- Generating python scripts using dnabot
```
Comment (shiltemann) [inline topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.md:17]: Could you rephrase these. They should be statements summing up the take-home messages. So ask yourself, if there are 3-5 things I hope the student will have learned from this, what is it?

So formulated not as "Doing task X", but "When doing X, always make sure to Y", or "Doing X is useful because Z" etc, real conclusions. The way these are phrased now, they are more objectives.

File: topics/synthetic-biology/slides/introduction.html [hunk 15]
```diff
@@ -0,0 +1,209 @@
...
+    - ioanagry
+    - jfaulon
+---
+
+# What is the goal of synthetic biology?
+
+The goal of **synthetic biology** is to **extend** or **modify** the behavior of organisms and **engineer** them to perform new tasks.
+
+.center[*LIFE AS IT SHOULD BE or AS WE WANT IT TO BE*]
+
```
[the remark this answers is not in the document]
Comment (kenza12) [reply topics/synthetic-biology/slides/introduction.html:21]: Hi @shiltemann , Thank you for your comments ! I think it's a good idea to add speaker notes but we're planning to do that later. Is it possible to release a first version of the tutorial and we can add more information or features later ?
The scientific content was double checked by my colleague.

File: topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.md [hunk 16]
```diff
@@ -0,0 +1,225 @@
...
+{: .hands_on}
+
+# Find enzymes using **Selenzyme**
+
+At first, a pathway generated by the **Pathway Analysis workflow** (https://galaxy-synbiocad.org) is provided as input to the Selenzyme tool ({% cite Carbonell2018 %}). Selenzyme searches for enzymes corresponding to each reaction of the pathway. It performs a reaction similarity search in the reference reaction database [MetaNetX](https://www.metanetx.org) and outputs an updated SBML file annotated with the enzyme UniProt IDs.
+The tool provides several scores that can be combined in order to define an overall score. Scores are given for reaction similarity, conservation based on a multiple sequence alignment, phylogenetic distance between source organism and host, and additional scores calculated from sequence properties as shown in the example bellow.
+
+![Selenzyme Results](../../images/selenzy_tool.png)
+
+> ### {% icon hands_on %} Hands-on: Task description
```
[the remark this answers is not in the document]
Comment (kenza12) [reply topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.md]: Ok, done, thanks.

File: topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.md [hunk 17]
```diff
@@ -0,0 +1,225 @@
...
+>    - *"Comma separated taxon IDs of output enzyme sequences"*: enter `553` which is the taxon ID of the Pantoea ananatis strain from which we want to extact enzymes
+>    - {% icon param-file %} *"Linkers and user parts": Select `parts_for_lycopene.csv` from your current history in csv format*
+>    - {% icon param-file %} *"DNA-Bot settings (Optionally)"*: Select either `dnabot_paris_settings.yaml` or `dnabot_london_settings.yaml` from your history)
+>
+>    > ### {% icon comment %} Comment
+>    >
+>    > All the outputs will be automatically generated and identical to the previous ones. 
+>    {: .comment}
+>
+
```
[the remark this answers is not in the document]
Comment (kenza12) [reply topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.md]: Ok, done, thanks.

File: topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.md [hunk 18]
```diff
@@ -0,0 +1,225 @@
...
+>
+{: .question}
+
+# Run the **Genetic Design - BASIC Assembly Workflow**
+
+In this section, you can run the Genetic Design - BASIC Assembly Workflow more easily and fastly following these instructions:
+
+> ### {% icon hands_on %} Hands-on: Task description
+>
+> 1. Import your **Genetic Design - Basic Assembly Workflow** by uploading the workflow file using the following **URL link** : https://training.galaxyproject.org/training-material/topics/synthetic-biology/tutorials/basic_assembly_analysis/workflows/Genetic_Design_BASIC_Assembly.ga.
```
[the remark this answers is not in the document]
Comment (kenza12) [reply topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.md]: Ok, done, thanks.

File: topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.md (hunk 13 above)
Comment (kenza12) [reply topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.md:47]: I submitted the video to the GTN Video Library 2 weeks ago. I don't know if you were able to analyze it ? Thanks.

File: topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.md (hunk 14 above)
Comment (kenza12) [reply topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.md]: Yes, understood, thank you for your explanation. We changed that.

File: topics/synthetic-biology/slides/introduction.html [hunk 19]
```diff
@@ -0,0 +1,209 @@
...
+.pull-right[<img src="../images/biobricks_logo.png" width="200"/>]
+.pull-left[
+- **the plasmid backbone**
+    - All plasmids in Parts Registry are in a standard **BioBricks** format which is a standard for interchangable parts, developed with a view to building biological systems in living cells
+]
+
+---
+## Synthetic biology approaches
+
+.pull-right[<img src="../images/circuit.png" width="120"/>]
```
Comment (shiltemann) [inline topics/synthetic-biology/slides/introduction.html:116]: you should be able to include images with markdown syntax here too, to specify the width, you should be able to do something like:


```
![alt text](path/to/image.png){: width="50%"}
```

(Not 100% sure if this also works in the slides)

Either way, please include alt text here as well (`alt="good description of image"` if you need to keep the html include)

That is the what is causing the last remaning linting errors, here and a handful more images in the slide deck

Thanks!

File: topics/synthetic-biology/slides/introduction.html (hunk 15 above)
[the remark this answers is not in the document]
Comment (shiltemann) [reply topics/synthetic-biology/slides/introduction.html:21]: yes, it won't prevent this from being merged, but would be great if you could add this in a future contribution :)

File: topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.md (hunk 13 above)
Comment (kenza12) [reply topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.md]: Hi @shiltemann , thank you for analyzing the video !

Since the video will appear from the top of the tutorial, I deleted the old video link in the tutorial.

About the speaker, no it's not me. He's one of the tutorial contributors:

```
breakthewall:
    name: Joan Hérisson
    email: joan.herisson@univ-evry.fr
    orcid: 0000-0001-9741-0847
    gitter: breakthewall
    linkedin: joanherisson
    joined: 2022-03
```
Name of institute: Université d'Évry / Paris-Saclay, CNRS, CEA, INRAE
Webpage: http://www.univ-evry.fr/

He's an employee of the University of Évry / Paris-Saclay supervised by CNRS and CEA. He's also working in collaboration with INRAE in this project. I will let you pick the needed informations.

Thanks again.

evry_logo
Logo-INRAE_Quadri-web
CNRS
CEA_logo_nouveau

Comment (shiltemann) [reply topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.md]: Perfect, thanks!

Here is a link to the video captions as generated by YouTube, if you could go over it fix where needed, that would be great

https://docs.google.com/document/d/1edwDeJtySnO2d9nLbwPBpoUeBdEYE-9sLKhCJiEJk3k/edit?usp=sharing

**Review by shiltemann — APPROVED** [review]

File: topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.md (hunk 13 above)
Comment (kenza12) [reply topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.md]: Ok, we fixed the video captions, thank you !!

**Comment by shiltemann** [comment]
Thanks @kenza12 and team!! This looks great, looking forward to seeing more synthetic biology tutorials added in the future!

**Comment by kenza12** [comment]
> Thanks @kenza12 and team!! This looks great, looking forward to seeing more synthetic biology tutorials added in the future!

That's great !! Many thanks to you @shiltemann and Galaxy team !

**Comment by shiltemann** [comment]
And it's live! :tada: :tada: 
https://training.galaxyproject.org/training-material/topics/synthetic-biology/

**Comment by kenza12** [comment]
I could see. Awesome !! Thanks again.
